common/src/main/java/org/opensearch/ml/common/controller/MLModelController.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.controller;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.common.utils.StringUtils.getParameterMap;
+
+@Data
+public class MLModelController implements ToXContentObject, Writeable {
+
+    public static final String MODEL_ID_FIELD = "model_id"; // mandatory
+    public static final String USER_RATE_LIMITER_CONFIG = "user_rate_limiter_config";
+
+    @Getter
+    private String modelId;
+    // The String is the username field where the MLRateLimiter is its corresponding rate limiter config.
+    private Map<String, MLRateLimiter> userRateLimiterConfig;
```

Comment:
In cases that multiple users have the same userRateLimiterConfig, or a single user has the same RateLimiter for multipleModels, do we need to send multiple redundant data to create or update the controller?

Reply:
Just want to check if there was any discussion to put the MLRateLimiter into the .plugins-ml-config system index. In this way you can store all the users as a set or list in one RateLimiter that are associated, and also you only need to use a RateLimiter id as reference in the MLModel. Isn't it redundant to have a Controller defined inside MLModel, and this controller also contains the same modelId?

Reply:
> In cases that multiple users have the same userRateLimiterConfig, or a single user has the same RateLimiter for multipleModels, do we need to send multiple redundant data to create or update the controller?
I am afraid we have to at this time. Maybe we can figure out a way as a feature enhancement in the future.

Reply:
`.plugins-ml-config` is not appropriate because it stored MASTER_KEY in it. So the access control will be difficult there. I think we did the similar discussion in our design review.

Reply:
Actually it shouldn't be that difficult. The ml-config system index has no public API built for it so no one can see it. But we can also build APIs that ensure this master_key is never visible.

Reply:
Oh I get you. Yeah I think this is doable, but we may take some additional effort to change the api and I think we should do some further discussion on any potential issues on this decision.